### PR TITLE
XYZ-52: Adapts 'Enable Team Creation' on 'Users and Teams' page to re…

### PR DESCRIPTION
**Pull request by mkraft**
…ad/update permissions.

#### Summary
Similar to the Policy page permissions changes, this change updates a permission on a role based on the value of the radio input labelled "Enable Team Creation" in System Console -> General -> Users and Teams.

#### Ticket Link
[XYZ-52](https://mattermost.atlassian.net/browse/XYZ-52)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx [hunk 1]
```diff
@@ -2,47 +2,107 @@
...
+import BooleanSetting from '../boolean_setting.jsx';
+import DropdownSetting from '../dropdown_setting.jsx';
+import SettingsGroup from '../settings_group.jsx';
+import TextSetting from '../text_setting.jsx';
 
 const RESTRICT_DIRECT_MESSAGE_ANY = 'any';
 const RESTRICT_DIRECT_MESSAGE_TEAM = 'team';
 
 export default class UsersAndTeamsSettings extends AdminSettings {
+    static propTypes = {
```
Comment (jespino) [inline components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx]: I miss the `roles` and `rolesRequest` declaration here.

File: components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx [hunk 2]
```diff
@@ -2,47 +2,107 @@
...
+        actions: PropTypes.shape({
+            loadRolesIfNeeded: PropTypes.func.isRequired,
+            editRole: PropTypes.func.isRequired
+        }).isRequired
+    };
+
     constructor(props) {
         super(props);
 
         this.getConfigFromState = this.getConfigFromState.bind(this);
```
Comment (jespino) [inline components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx]: We can convert it to an arrow function and remove this line. (Not directly related to the PR)

File: components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx [hunk 3]
```diff
@@ -2,47 +2,107 @@
...
 
         this.renderSettings = this.renderSettings.bind(this);
```
Comment (jespino) [inline components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx]: We can convert it to an arrow function and remove this line. (Not directly related to the PR)

Comment (mkraft) [reply components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx]: @jespino Yes good callouts. Might as well make improvements while we're here.

**Comment by mkraft** [comment]
@jespino Updated.

**Pull request merged**